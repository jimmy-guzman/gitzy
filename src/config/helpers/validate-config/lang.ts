import { validIssuesPrefixes } from './constants'

export const lang = {
  breakingChangeEmoji: 'breakingChangeEmoji must be a string',
  closedIssueEmoji: 'closedIssueEmoji must be a string',
  details:
    'details must look like "{ description: "A new feature", emoji: "âœ¨" }"',
  disableEmoji: 'disableEmoji must be a boolean',
  headerMaxLength: 'headerMaxLength must be a number',
  headerMinLength: 'headerMinLength must be a number',
  issuesPrefix: `issuesPrefix must be one of ${validIssuesPrefixes.join(', ')}`,
  questions: 'questions must be an array of strings',
  scopes: 'scopes must be an array of strings',
  types: 'types must be an array of strings',
  useCommitlintConfig: 'useCommitlintConfig must be a boolean',
}
