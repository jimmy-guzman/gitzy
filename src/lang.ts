import type { Flags } from "./interfaces";

interface Lang {
  description: string;
  examples: string;
  flags: Record<keyof Omit<Flags, "emoji" | "help" | "version">, string>;
}

const skipQuestionMessage = (question: string) => {
  return `skip "${question}" question and provide your own "${question}" message`;
};

export const lang = {
  description: "interactive conventional commits cli",
  examples: `
  $ gitzy
  $ gitzy -p -a
  $ gitzy -m "added cool new feature" -t "feat" -s "amazing"
  $ gitzy -lD --no-emoji
  `,
  flags: {
    body: skipQuestionMessage("body"),
    breaking:
      'mark as breaking change. Pass a message for "footer"/"both" formats, or just the flag for "!" format',
    commitlint: "leverage commitlint's configuration",
    dryRun: "displays git message but does not commit",
    hook: "run gitzy inside a Git hook",
    issues: skipQuestionMessage("issues"),
    noEmoji: "disable all emojis",
    passthrough: 'subsequent command line args passed through to "git"',
    retry: "retries previous commit, skips all prompts",
    scope: skipQuestionMessage("scope"),
    skip: "skip questions",
    subject: skipQuestionMessage("subject"),
    type: skipQuestionMessage("type"),
  },
} satisfies Lang;
