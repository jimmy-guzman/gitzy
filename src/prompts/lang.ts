import { bold, dim, red, reset } from 'ansi-colors'

import type { IssuesPrefixes, PromptsLang } from '../interfaces'

const breaking = red('BREAKING CHANGE:')

const closes = (issuesPrefix: IssuesPrefixes): string => {
  return reset(`${issuesPrefix}:`)
}

export const promptsLang: PromptsLang = {
  body: {
    hint: '...supports multi line, press enter to go to next line',
    message: 'Add a longer description\n',
  },
  breaking: {
    hint: dim('...skip when none'),
    message: `${bold('Add any breaking changes')}\n  ${breaking}`,
  },
  scope: {
    hint: dim('...type or use arrow keys'),
    message: 'Choose the scope',
  },
  subject: { message: 'Add a short description' },
  type: {
    hint: dim('...type or use arrow keys'),
    message: 'Choose the type',
  },
}

export const issuesMessage = (issuesPrefix: IssuesPrefixes): string => {
  return `${bold('Add issues this commit closes, e.g #123')}\n  ${closes(
    issuesPrefix
  )}`
}

export const errorMessage = {
  minTitleLength: (length: number): string =>
    `The subject must have at least ${length} characters`,
  maxTitleLength: (length: number): string =>
    `The subject must be less than ${length} characters`,
}
