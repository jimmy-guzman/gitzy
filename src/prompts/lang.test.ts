import { promptsLang } from './lang'

describe('lang', () => {
  describe('promptsLang', () => {
    it('should create prompts lang', () => {
      expect(promptsLang).toMatchSnapshot()
    })
  })
})
