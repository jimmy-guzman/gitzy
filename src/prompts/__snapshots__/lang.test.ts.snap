// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lang > promptsLang > should create prompts lang 1`] = `
{
  "body": {
    "hint": "...supports multi line, press enter to go to next line",
    "message": "Add a longer description
",
  },
  "breaking": {
    "hint": "[2m...skip when none[22m",
    "message": "[1mAdd any breaking changes[22m
  [31mBREAKING CHANGE:[39m",
  },
  "scope": {
    "hint": "[2m...type or use arrow keys[22m",
    "message": "Choose the scope",
  },
  "subject": {
    "message": "Add a short description",
  },
  "type": {
    "hint": "[2m...type or use arrow keys[22m",
    "message": "Choose the type",
  },
}
`;
